http-api-monitor-offers.adoc

```diff
@@ -0,0 +1,327 @@
+
+[source,json]
+----
+{
+    "offers": [OfferDetail],
+    "total": integer
+}
+----
+
+Where `offers` is an array with individual offers and `total` is number of all offers. The model for each individual `OfferDetail` is defined as follows:
+
+[source,json]
+----
+{
+    acceptedBankIds	[string]
+    acceptedCountryCodes	[string]
+    amount	integer($int64)
+    arbitratorNodeAddresses	[string]
+    bankId	string
+    baseCurrencyCode	string
+    blockHeightAtOfferCreation	integer($int64)
+    buyerSecurityDeposit	integer($int64)
+    counterCurrencyCode	string
+    countryCode	string
+    currencyCode	string
+    date	string($date-time)
+    direction	string Enum: [ BUY, SELL ]
+    hashOfChallenge	string
+    id	string
+    isCurrencyForMakerFeeBtc	boolean
+    isPrivateOffer	boolean
+    lowerClosePrice	integer($int64)
+    makerFee	integer($int64)
+    makerPaymentAccountId	string
+    marketPriceMargin	number($double)
+    maxTradeLimit	integer($int64)
+    maxTradePeriod	integer($int64)
+    minAmount	integer($int64)
+    offerFeePaymentTxId	string
+    ownerNodeAddress	string
+    paymentMethodId	string
+    price	integer($int64)
+    protocolVersion	integer($int32)
+    sellerSecurityDeposit	integer($int64)
+    state	string Enum: [ UNKNOWN, OFFER_FEE_PAID, AVAILABLE, NOT_AVAILABLE, REMOVED, MAKER_OFFLINE ]
+    txFee	integer($int64)
+    upperClosePrice	integer($int64)
+    useAutoClose	boolean
+    useMarketBasedPrice	boolean
+    useReOpenAfterAutoClose	boolean
+    versionNr	string
+}
+----
+
+Now let's assume that we want to buy bitcoin, and let's define "interesting" offers as those that:
+
+ . sell bitcoin at a discount 1% or more under the current market price, and
+ . accept payment in Euros to a Polish SEPA account
+
+First we need to filter those offers using following static criteria:
+
+[source,json]
+----
+{
+    baseCurrencyCode: 'BTC',
+    counterCurrencyCode: 'EUR',
+    direction: 'SELL',
+    paymentMethodId: 'SEPA'
+}
+----
+
+Next we need to filter those offers by price. There are two types of offers: _market-based price offers_ and _fixed price offers_. They are distinguished by the `useMarketBasedPrice` attribute. In case of market-based price offers the filtering criteria is easy: the `marketPriceMargin` value must be above 0.1. In case of fixed price offers we have to fetch market price of BTC in EUR, then calculate whether the price is 1% or less, and finally we must filter offers which have a price _less_ than that calculated price.
+
+=== Get the market price
+
+In order to get the market price of BTC in EUR, execute the following query:
+
+    curl http://localhost:8080/api/v1/currencies/prices?currencyCodes=EUR
+
+You should receive a response like the following:
+
+[source,json]
+----
+{
+  "prices": {
+    "EUR": 7035.62
+  }
+}
+----
+
+== The JavaScript part
+
+Let's install some dependencies:
+
+    npm install lodash http-as-promised nodemailer
+
+In general our script will look like this:
```

Comment:
In the sections that follow, I recommend that all source code snippets:

- [ ] get sourced from an actual, e.g. 'bisqmon.js' source file checked into the repository. Or perhaps `http-api-monitor-offers.js` to align with the naming of this file. We'll probably need to think more about how to structure companion sources along with these docs in a flat namespace, but it doesn't matter much at this point: what does matter is that the code below should get extracted as snippets from real source code that actually works. Asciidoctor has great tools to help here, see https://asciidoctor.org/docs/user-manual/#include-partial.

 - [ ] get labeled such that they show the name of the file they're coming from (it's fine if all three are labeled as coming from the same file. The point is that this guide needs to be an extremely literal, step-by-step process of going from zero to working code. So if there is code that the user should write, you should tell them where to write it (i.e. in a file named `bisqmon.js`) and in which order to write it. Add this statement, add this function, and that function, and you're done.

Reply:
both done